### PR TITLE
multiple room functionality for training bonsai brain

### PR DESCRIPTION
**Pull request by jillianmclements**
- moved "determine_room_parameters" to inside the building subsystem within the sim
- \# rooms and # windows are now config variables, along with Toutdoor

### CODE REVIEW

File: samples/building_energy_management/buildingEnergyManagement.ink [hunk 1]
```diff
@@ -40,44 +54,66 @@ function TempDiff(Tin:number, Tset:number) {
 graph (input: ObservableState): SimAction {
     concept adjust(input): SimAction {
         curriculum {
-            source simulator (action: SimAction, config: SimConfig): ObservableState {
+            source simulator (action: SimAction, config: SimConfig): SimState {
                 # package "bem_final"
```
Comment (journeyman-msft) [inline samples/building_energy_management/buildingEnergyManagement.ink:58]: please remove package comment

File: samples/building_energy_management/buildingEnergyManagement.ink [hunk 2]
```diff
@@ -40,44 +54,66 @@ function TempDiff(Tin:number, Tset:number) {
...
+            goal (State: SimState) {
                 minimize `Temp Deviation`:
-                    TempDiff(State.Troom1, State.Tset) in Goal.RangeBelow(MaxDeviation)
+                    TempDiff(TransformState(State).Troom_avg, TransformState(State).Tset) in Goal.RangeBelow(MaxDeviation)
             }
 
             lesson adjust_easy {
                 scenario {
                     input_Toutdoor: 60.0,
+                    input_nRooms: 3,
```
Comment (journeyman-msft) [inline samples/building_energy_management/buildingEnergyManagement.ink:77]: could we leave nrooms = 1 for starting

File: samples/building_energy_management/init_vars.m [hunk 3]
```diff
@@ -32,6 +21,4 @@
 TinIC = 20;
 
 %% Air conditioner
-
-dTac = 20*(5/9); % deg C
```
Comment (journeyman-msft) [inline samples/building_energy_management/init_vars.m:36]: Can we keep this and add it as a variable in the sim? I know we have 100/9 -- I find it difficult to follow without units , this is more clear for new users

File: samples/building_energy_management/run_benchmark.m [hunk 4]
```diff
@@ -5,6 +5,11 @@
 clc;
 
 %% Initalize Parameters
+n_rooms =  3;
```
Comment (journeyman-msft) [inline samples/building_energy_management/run_benchmark.m:8]: default to 1

**Review by journeyman-msft — APPROVED** [review]
Please address comments and then merge

File: samples/building_energy_management/init_vars.m (hunk 3 above)
Comment (jillianmclements) [reply samples/building_energy_management/init_vars.m:36]: Agreed! Good catch.

**Pull request merged**